Sources/BraveWallet/Crypto/Stores/NFTStore.swift

```diff
@@ -520,29 +495,60 @@ public class NFTStore: ObservableObject, WalletObserverStore {
   
   private func buildNFTGroupModels(
     groupBy: GroupBy,
-    allUserNFTs: [NetworkAssets],
+    spams: [NetworkAssets],
     selectedAccounts: [BraveWallet.AccountInfo],
     selectedNetworks: [BraveWallet.NetworkInfo]
-  ) -> [NFTGroupViewModel] {
+  ) -> ([NFTGroupViewModel], [BraveWallet.BlockchainToken]) {
+    
+    // user visible NFTs
+    let userVisibleNFTs = assetManager.getAllUserAssetsInNetworkAssetsByVisibility(networks: selectedNetworks, visible: true)
+      .map { networkAssets in
+        NetworkAssets(
+          network: networkAssets.network,
+          tokens: networkAssets.tokens.filter { $0.isNft || $0.isErc721 },
+          sortOrder: networkAssets.sortOrder
+        )
+      }
+    // user hidden NFTs
+    let userHiddenNFTs = assetManager.getAllUserAssetsInNetworkAssetsByVisibility(networks: selectedNetworks, visible: false)
+      .map { networkAssets in
+        NetworkAssets(
+          network: networkAssets.network,
+          tokens: networkAssets.tokens.filter { $0.isNft || $0.isErc721 },
+          sortOrder: networkAssets.sortOrder
+        )
+      }
+    
+    let allUserNFTsInNetworks = generateAllNFTsInNetworks(
+      userVisibleNFTs: userVisibleNFTs,
+      userHiddenNFTs: userHiddenNFTs,
+      computedSpamNFTs: spams
+    )
+    
+    var allUserNFTs: [BraveWallet.BlockchainToken] = []
+    for networkAssets in [userVisibleNFTs, userHiddenNFTs, spams] {
+      allUserNFTs.append(contentsOf: networkAssets.flatMap(\.tokens))
+    }
```

Comment:
Could just be 
`let allUserNFTs = (userVisibleNFTs + userHiddenNFTs + spams).flatMap(\.tokens)`